### PR TITLE
ouiRadio

### PR DESCRIPTION
**Pull request by neolitec**
Signed-off-by: Kevin Manson kevin.manson@corp.ovh.com

https://ovh_ui_angular-angular_oui_radio.uxci.ovh/#!/oui-angular/radio

### CODE REVIEW

**Comment by jimmyfortinx** [comment]
(Bug) In your **Basic** example the *Value A* isn't selected and when I clicked on *Value B* it select *Value A*.

File: packages/oui-radio/README.md [hunk 1]
```diff
@@ -4,96 +4,79 @@
...
+## API
+
+| Attribute     | Type                    | Binding | One-time Binding | Values                   | Default | Description
+| ----          | ----                    | ----    | ----             | ----                     | ----    | ----
+| text          | string                  | @       |                  |                          |         | checkbox text
+| description   | string                  | @?      |                  |                          |         | description text
+| id            | string                  | @?      | `true`           |                          |         | id attribute of the checkbox
+| name          | string                  | @?      | `true`           |                          |         | name attribute of the checkbox
+| disabled      | boolean                 | <?      |                  |                          | false   | disabled flag
+| model         | Object                  | =?      |                  |                          |         | current value of the checkbox and null is considered as `indeterminate`
```
Comment (jimmyfortinx) [inline packages/oui-radio/README.md]: You're still talking about the *indeterminate* state, but radio didn't have one.

File: packages/oui-radio/src/radio.controller.js [hunk 2]
```diff
@@ -1,76 +1,96 @@
...
-            this.checked = true;
+        if (!self.id) {
+            this.id = `oui-radio-${this.$scope.$id}`;
         }
+    }
 
-        if (_.has(this.$attrs, "disabled") && _.isEmpty(this.$attrs.disabled)) {
-            this.disabled = true;
-        }
+    // constructor ($scope, $element, $attrs, $transclude) {
```
Comment (jimmyfortinx) [inline packages/oui-radio/src/radio.controller.js]: Did you forget to remove those comments ?

File: packages/oui-radio/src/radio.controller.js [hunk 3]
```diff
@@ -1,76 +1,96 @@
...
-        const self = this;
-
-        this.onGroupValueChange(self.group.value);
+    $postLink () {
+        this.$element.removeAttr(["id", "name"]);
 
-        this.$element.find(INPUT_SELECTOR).bind("change", function () {
-            self.$scope.$apply(() => self.group.callOnChange(this.value));
-        });
+        this.checkboxElement = this.$element.find("input");
```
Comment (jimmyfortinx) [inline packages/oui-radio/src/radio.controller.js]: `checkboxElement` inside radio ;). Anyway you do not use it.

File: packages/oui-radio/src/index.spec.js [hunk 4]
```diff
@@ -1,24 +1,179 @@
...
+            it("should display a text inside the radio's text container", () => {
+                const element = TestUtils.compileTemplate('<oui-radio text="test"></oui-radio>');
+
+                const textContainerRadioElement = getRadioTextContainerElement(element);
+                expect(angular.element(textContainerRadioElement).html()).toBe("test");
+            });
+        });
+
+        describe("description attribute", () => {
+            it("should display the radio's description container when empty", () => {
```
Comment (tremblaymath) [inline packages/oui-radio/src/index.spec.js]: I think it should be "should not display a text inside the radio's description container" to match the assertion

**Pull request merged**